components/json_path.rst

```diff
@@ -0,0 +1,268 @@
+
+Accessing a Specific Property
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Use dot-notation for object keys and square brackets for array indices. The root
+of the document is represented by ``$``::
+
+    // Get the title of the first book in the store
+    $titles = $crawler->find('$.store.book[0].title');
+
+    // $titles is ['Sayings of the Century']
+
+While dot-notation is common, JSONPath offers alternative syntaxes which can be more flexible. For instance, bracket notation (`['...']`) is required if a key contains spaces or special characters.
+
+::
+
+    // This is equivalent to the previous example and works with special keys
+    $titles = $crawler->find('$["store"]["book"][0]["title"]');
+
+    // You can also build the query programmatically for more complex scenarios
+    use Symfony\Component\JsonPath\JsonPath;
+
+    $path = (new JsonPath())->key('store')->key('book')->first()->key('title');
+    $titles = $crawler->find($path);
+
+Searching with the Descendant Operator
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The descendant operator (``..``) recursively searches for a given key, allowing
+you to find values without specifying the full path.
+
+::
+
+    // Get all authors from anywhere in the document
+    $authors = $crawler->find('$..author');
+
+    // $authors is ['Nigel Rees', 'Evelyn Waugh', 'Herman Melville', 'John Ronald Reuel Tolkien']
+
+Filtering Results
+~~~~~~~~~~~~~~~~~
+
+JSONPath includes a powerful filter syntax (``?(<expression>)``) to select items
+based on a condition. The current item within the filter is referenced by ``@``.
+
+::
+
+    // Get all books with a price less than 10
+    $cheapBooks = $crawler->find('$.store.book[?(@.price < 10)]');
+
+    /*
+    $cheapBooks contains two book objects:
+    the one by "Nigel Rees" and the one by "Herman Melville"
+    */
+
+Building Queries Programmatically
+---------------------------------
+
+For more dynamic or complex query building, you can use the fluent API provided
+by the :class:`Symfony\\Component\\JsonPath\\JsonPath` class. This allows you
+to construct a query object step-by-step. The ``JsonPath`` object can then be
+passed to the crawler's :method:`Symfony\\Component\\JsonPath\\JsonCrawler::find` method.
+
+The main advantage of the programmatic builder is that it automatically handles
+the correct escaping of keys and values, preventing syntax errors.
+
+::
+
+    use Symfony\Component\JsonPath\JsonPath;
+
+    $path = (new JsonPath())
+        ->key('store')       // Selects the 'store' key
+        ->key('book')        // Then the 'book' key
+        ->index(1);          // Then the item at index 1 (the second book)
+
+    // The created $path object is equivalent to the string '$["store"]["book"][1]'
+    $book = $crawler->find($path);
+
+    // $book contains the book object for "Sword of Honour"
+
+The ``JsonPath`` class provides several methods to build your query:
+
+``key(string $name)``
```

Comment:
If I remember correctly, you should use the list notation using `*`. You can find many places where it is done this way in the other pages of the doc

Reply:
Fixed